irobotclient/main.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+import errno
+
+from os import path
+from requests import Response
+
+from irobotclient import configuration_handler
+from irobotclient import request_formatter
+from irobotclient.custom_exceptions import IrobotClientException
+from irobotclient.request_handler import Requester
+
+
+def _print_error_details(error: OSError):
+    """
+    Print the output of any errors raised.
+
+    Utilises the OSError exception class which the iRobot Client uses for custom exceptions too.
+
+    :param error:
+    :return:
+    """
+    print("Inside _output_error_details") # Beth - debug
+    print(error)
+    exit(error.errno)
+
+
+def _download_data(response: Response, output_dir:str):
+    """
+
+    :param response:
+    :param output_dir:
+    :return:
+    """
+    try:
+        file_name = (path.split(response.url))[1]
```

Comment:
What if you're downloading `/path/to/project-a/README` and `/path/to/project-b/README`? It's true that CRAM files will _probably_ have unique names, but I don't even think that is a sure thing. You need to account for such conflicts because, when they happen -- they _will_ happen! -- with the current behaviour it will be very difficult to understand what is going wrong to an end user.